eo-runtime/src/main/eo/org/eolang/negative-infinity.eo

```diff
@@ -59,48 +59,78 @@
     ^.eq x > @
 
   # Multiplication of $ and x
-  [x] > times
+  [x...] > times
+    x.at 0 > first!
     if. > @
       or.
-        x.as-bytes.eq (nan.as-bytes)
-        x.eq 0.0
-        x.eq 0
+        first.as-bytes.eq (nan.as-bytes)
+        first.eq 0.0
+        first.eq 0
       nan
       if.
-        x.eq ^
+        first.eq ^
         positive-infinity
         ^
 
   # Sum of $ and x
-  [x] > plus
+  [x...] > plus
+    positive-infinity.as-bytes > pos-inf-as-bytes!
+
+    [acc index terms check-term] > plus-rec
+      if. > @
+        index.eq terms.length
+        acc
+        if.
+          check-term
+            terms.at index
+          nan
+          plus-rec
+            acc
+            index.plus 1
+            terms
+            check-term
+
+    [term] > check-term
+      term > t!
+      or. > @
+        not.
+          t.eq t
+        t.as-bytes.eq pos-inf-as-bytes
+
     if. > @
-      or.
-        x.as-bytes.eq (nan.as-bytes)
-        x.eq positive-infinity
-      nan
+      eq.
+        x.length > terms-count!
+        0
       ^
+      if.
+        terms-count.gt 1
+        plus-rec
+          ^
+          0
+          x
+          check-term
+        if.
+          check-term
+            x.at 0
+          nan
+          ^
 
   # Negation of $
   [] > neg
     positive-infinity > @
 
   # Difference between $ and x
-  [x] > minus
-    ^.plus (x.neg) > @
-
-    [x] > not-int
-      x.as-int > as-integer!
-      and. > @
-        as-integer.as-float.lt x
-        (as-integer.plus 1).as-float.gt x
+  [x...] > minus
+    ^.plus ((x.at 0).neg) > @
```

Comment:
@maxonfjvipon Can you explain why we pass to `plus` object only the first element of the `x...` vararg array?

Reply:
@Graur I added 3 pdd puzzles in `positive-infinity.eo` to implement these objects in the future: minus, times, div. `x...` in these objects are kind of placeholders now

Reply:
@maxonfjvipon Yeah, I missed that, thanks!